cedar-policy-core/src/entities/json/entities.rs

```diff
@@ -212,6 +213,39 @@ impl<'e, 's, S: Schema> EntityJsonParser<'e, 's, S> {
         Ok(entities.into_iter())
     }
 
+    /// Parse a single entity from an in-memory JSON value
+    pub fn single_from_json_value(
+        &self,
+        value: serde_json::Value,
+    ) -> Result<Entity, EntitiesError> {
+        let ejson: EntityJson =
+            serde_json::from_value(value).map_err(JsonDeserializationError::from)?;
+        let entity = self.parse_ejson(ejson)?;
+        match self.schema {
+            Some(schema) => {
+                let checker = crate::entities::conformance::EntitySchemaConformanceChecker::new(
+                    schema,
+                    self.extensions,
+                );
+                checker.validate_entity(&entity)?;
+                Ok(entity)
+            }
+            None => Ok(entity),
+        }
+    }
+
+    /// Parse a single entity from a JSON string
+    pub fn single_from_json_str(&self, src: impl AsRef<str>) -> Result<Entity, EntitiesError> {
+        let v = serde_json::from_str(src.as_ref()).map_err(JsonDeserializationError::from)?;
+        self.single_from_json_value(v)
+    }
```

Comment:
This organization results in two calls to `serde_json` -- one `from_str()` and then another `from_value()`.  We could avoid this by making a helper function that takes an `EntityJson` and does the rest.  All three variants would call a single `serde_json` function and then that helper.